lib/hyacinth/digital_object/type_def/dynamic_field_data.rb

```diff
@@ -73,43 +60,13 @@ def to_serialized_form_impl(json_var)
           json_var = json_var.deep_dup
 
           # Dehydrate URI terms. Remove all other fields in terms hash except for 'uri'
-          extract_terms(field_map, json_var).values.sum([]).each { |t| t.slice!('uri') }
+          map.extract_terms(json_var).values.sum([]).each { |t| t.slice!('uri') }
 
           json_var
         end
 
-        def field_map
-          Hyacinth::DynamicFieldsMap.generate(*TYPE_TO_FORM_TYPE[@type])
-        end
-
-        def extract_terms(map, data)
-          terms = {}
-
-          data.each do |field_or_group_key, value|
-            next unless map.key?(field_or_group_key)
-
-            reduced_map = map[field_or_group_key]
-
-            case reduced_map[:type]
-            when 'DynamicFieldGroup'
-              next unless value.is_a?(Array)
-
-              value.each do |v|
-                extract_terms(reduced_map[:children], v).each do |vocab, new_terms|
-                  terms[vocab] = terms.fetch(vocab, []).concat(new_terms)
-                end
-              end
-            when 'DynamicField'
-              next unless reduced_map[:field_type] == DynamicField::Type::CONTROLLED_TERM
-              next unless value.is_a?(Hash)
-              vocab = reduced_map[:controlled_vocabulary]
-
-              terms[vocab] = [] unless terms.key?(vocab)
-              terms[vocab] += [value]
-            end
-          end
-
-          terms
+        def map
```

Comment:
Change this to `field_map` as to not confused with `Enumerable#map`

Reply:
Thanks!